Portal : make markdown scroll work again

ThibaudAV opened the pull request:
## Issue
https://gravitee.atlassian.net/browse/APIM-3582
https://gravitee.atlassian.net/browse/APIM-3604

## Description

- bump maked to last version
- fix scroll 
- fix click into ToC 
- on page refresh try to scroll to right anchor 
- move sanitize added for translation page into backend

- also fix empty state with single api into Portal

## Additional context

<!-- Add any other context about the PR here -->
<!-- It can be links to other PRs or docs or drawing -->
<!-- Or reproduction steps in case of bug fix -->

<!-- Environment placeholder -->

🏗️ Your changes can be tested here and will be available soon:
      Console: [https://pr.team-apim.gravitee.dev/6230/console](https://pr.team-apim.gravitee.dev/6230/console)
      Portal: [https://pr.team-apim.gravitee.dev/6230/portal](https://pr.team-apim.gravitee.dev/6230/portal)
      Management-api: [https://pr.team-apim.gravitee.dev/6230/api/management](https://pr.team-apim.gravitee.dev/6230/api/management)
      Gateway v4: [https://pr.team-apim.gravitee.dev/6230](https://pr.team-apim.gravitee.dev/6230)
      Gateway v3: [https://pr.gateway-v3.team-apim.gravitee.dev/6230](https://pr.gateway-v3.team-apim.gravitee.dev/6230)

<!-- Environment placeholder end -->
<!-- Storybook placeholder -->
---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-fwultzqiux.chromatic.com)
<!-- Storybook placeholder end -->

ThibaudAV committed:
chore(console): bump ui-components

ThibaudAV committed:
fix(portal): make markdown scroll work again

Move translation sanitize into backend : https://gravitee.atlassian.net/browse/APIM-2012

https://gravitee.atlassian.net/browse/APIM-3582

phiz71 commented on gravitee-apim-portal-webui/src/app/app.module.ts:
```diff
@@ -151,6 +151,7 @@ export class AppModule {
         // backward navigation
         viewportScroller.scrollToPosition(e.position);
       } else if (!e.anchor) {
+        console.log(e);
```
to remove

phiz71 commented on gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/PageService.java:
```diff
@@ -16,6 +16,7 @@
 package io.gravitee.rest.api.service;
 
 import io.gravitee.common.data.domain.Page;
+import io.gravitee.repository.exceptions.TechnicalException;
```
?

phiz71 commented on gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/PageServiceImpl.java:
```diff
@@ -2511,7 +2511,21 @@ private List<String> validateSafeContent(ExecutionContext executionContext, Page
-            if (markdownSanitize && PageType.MARKDOWN.name().equals(pageEntity.getType())) {
+            var pageType = pageEntity.getType();
+
+            if (pageEntity.getParentId() != null && PageType.TRANSLATION.name().equals(pageType)) {
```
Good catch !

phiz71 reviewed (approved):
LGTM except for the 2 small comments

ThibaudAV committed:
fix(portal): do not display empty state with one api

updateEmptyState is called by the 2 concurrent promises and will update the empty state regardless of the resolution order.
https://gravitee.atlassian.net/browse/APIM-3604
